week2/week2.py

```diff
@@ -0,0 +1,26 @@
+class MatchParentheses:
+
+    def __init__(self, base: str, target: str):
+        self.base = base
+        self.target = target
+
+    def match(self) -> int:
```

Comment:
If you're going to make this a class with an init and a method without args, might be nice to cache the result so if we call match multiple times it doesn't run the algo.

e.g. in `__init__` do `self.min_insertions = None` then in match at the start do

```
if self.min_insertions is not None:
  return self.min_insertions
```
and then update the `min_insertions` refs in match to `self.min_insertions`

Reply:
The alternative is to just remove the `__init__` method and have `match` take the base and target.